### DISCUSSION TITLE
Windows installation and running experimental scripts

### DISCUSSION THREAD

**Discussion by ramyma** [ramyma/a8r8#5] [General]
I added a couple of bat scripts to install/update and run the application on a Windows machine.

You'll need to:
1. Clone the repo using git `git clone https://github.com/ramyma/a8r8.git`
2. Run https://github.com/ramyma/a8r8/blob/main/install.ps1 to install
3. Run https://github.com/ramyma/a8r8/blob/main/run.ps1 to run

Keep in mind I'll need some help from the community to test it on Windows machines. 
Please report any issues here.

`Edit: Make sure to run it within a power shell window to capture any helpful messages.
`

**Comment by MaisonMeta** [comment]
Install.bat is not loading - the terminal opens and close automatically

**Reply by ramyma** [reply]
@MaisonMeta  I pushed an update to the install script.
**Please make sure to run it within a power shell window rather than running the file directly**

**Comment by MaisonMeta** [comment]
redownloaded -> 
![f0983cd615924cbc06b450939deefa6d](https://github.com/ramyma/a8r8/assets/142212033/7610919f-f8dd-4389-a454-725a5a861498)

Then launching run.bat -> 
'mix' is not recognized as an internal or external command,
operable program or batch file.

**Comment by ghost** [comment]
I've exactly the same problem as MaisonMeta. Trying docker installation instead now and will give feedback if I experimentet a bit with it.

Btw: Definitely keep working on this, from first impressions this could be something really big. If you're able to combine the freedom and features of ComfyUI + A1111 into a one-click installation with good UI/UX and auto-download of all essential models and stuff, this thing will go through the roof. Because thats one of the key things local SD is currently lacking.

**Comment by ramyma** [comment]
After doing some research it seems that one of the dependencies is not supported on Windows at the moment.
I added a one click installer for docker on Windows and updated the other scripts as well, in case a future fix was introduced.

https://github.com/ramyma/a8r8?tab=readme-ov-file#docker

https://github.com/ramyma/a8r8/blob/main/install-docker.ps1

**Comment by ramyma** [comment]
I added a new installation method for [Windows](https://github.com/ramyma/a8r8?tab=readme-ov-file#windows-wsl2-recommended) with the latest release [v0.4.0](https://github.com/ramyma/a8r8/releases/tag/v0.4.0).

Let me know if you need help or if the steps need to be updated.

**Comment by ramyma** [comment]
If anyone is stuck with the installation steps on  Windows WSL2, I pushed an update that fixes some of the known issues.

I noticed that cloning the repo on Windows causes some formatting issues with the installation and run scrpts when cloned from git in Windows.

Cloning the repo in a wsl session with 
```
wsl
git clone https://github.com/ramyma/a8r8.git
```
fixes the issue.

I'm working on a refined installation script that will cover the cloning part as well to make it easier for everyone.

**Comment by ramyma** [comment]
I pushed a new one click installer for Windows:
https://github.com/ramyma/a8r8?tab=readme-ov-file#one-click-windows-wsl2-recommended

This is now the recommended installation method for Windows.
